transcoders.go

```diff
@@ -292,13 +293,34 @@ func garlic32Validate(b []byte) error {
 
 var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS, p2pVal)
 
+// The encoded peer ID can either be a CID of a key or a raw multihash (identity
+// or sha256-256).
 func p2pStB(s string) ([]byte, error) {
-	// the address is a varint prefixed multihash string representation
-	m, err := mh.FromB58String(s)
+	// check if the address is a base58 encoded sha256 or identity multihash
+	if strings.HasPrefix(s, "Qm") || strings.HasPrefix(s, "1") {
+		m, err := mh.FromB58String(s)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse p2p addr: %s %s", s, err)
+		}
+		return m, nil
+	}
+
+	// check if the address is a CID
+	c, err := cid.Decode(s)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse p2p addr: %s %s", s, err)
 	}
-	return m, nil
+
+	ty := c.Type()
+	if ty == cid.Libp2pKey {
+		return c.Hash(), nil
+	}
+
+	codecStr := cid.CodecToStr[ty]
+	if codecStr == "" {
+		codecStr = fmt.Sprintf("[unknown multicodec %d]", ty)
+	}
```

Comment:
I copied this from go-libp2p-core, but is this overkill?

Reply:
Yeah, this seems like overkill. We can just fail.

Reply:
removed, look good?